src/analyses/everMultiThreaded.ml

```diff
@@ -0,0 +1,47 @@
+(** Work in progress *)
+
+open Analyses
+
+module UnitV =
+struct
+  include Printable.Unit
+  include StdV
+end
+
+module Spec : Analyses.MCPSpec =
+struct
+
+  (** Provides some default implementations *)
+  include Analyses.IdentitySpec
+
+  let name () = "evermultithreaded"
+
+  module D = Lattice.Unit
+  module C = D
+  module V = UnitV
+  module G = BoolDomain.MayBool
+
+  let startstate _ = ()
+  let exitstate = startstate
+
+  (** Sets the global invariant to true when a thread is spawned *)
+  let threadspawn ctx lval f args fctx =
+    ctx.sideg () true;
+    ()
+
+  let query ctx (type a) (q: a Queries.t) : a Queries.result =
+    match q with
+    | Queries.IsEverMultiThreaded ->
+      (match ctx.global () with
```

Comment:
Do you query `IsEverMultiThreaded` anywhere?

Reply:
We have enabled it now.

Reply:
It's used by then loop termination analysis.